### PR TITLE
chore: bump helm versions

### LINKED ISSUE #747: Bump observability helm charts
We are consuming 7.10.0 version, but 7.10.2 is out

cc/ @jmlrt

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
It bumps versions for Helm test suite:

- Elastic Helm charts: from 7.10.0 to 7.11.1
- Helm: from 3.4.1 to 3.5.2
- Kind: from 0.8.1 to 0.10.0

We also found that filebeat uses a daemonset for deployment, exactly as metricbeat does. We updated the code that calculated the resource name to match the new behaviour. For further information, please see [the original PR that added that change](https://github.com/elastic/helm-charts/pull/964).

## Why is it important?
Test latest versions

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the Unit tests for the CLI, and they are passing locally
- [x] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

## Author's Checklist

- [ ] @jmlrt are versions correct?

## How to test this PR locally

```shell
SUITE="helm" DEVELOPER_MODE=false TIMEOUT_FACTOR=1 LOG_LEVEL=TRACE ELASTIC_APM_ACTIVE=true make -C e2e functional-test
```

## Related issues

- Closes #747

### CODE REVIEW

**Review by cachedout — APPROVED** [review]

File: e2e/_suites/helm/helm_charts_test.go [hunk 1]
```diff
@@ -362,7 +362,7 @@ func (ts *HelmChartTestSuite) getResourceName(resource string) string {
 	} else if resource == k8s.ResourceTypes.ClusterRoleBinding {
 		return strings.ToLower(ts.Name + "-" + ts.Name + "-cluster-role-binding")
 	} else if resource == k8s.ResourceTypes.ConfigMap {
-		if ts.Name == "metricbeat" {
+		if ts.Name == "filebeat" || ts.Name == "metricbeat" {
```
Comment (mdelapenya) [inline e2e/_suites/helm/helm_charts_test.go:365]: @jmlrt I checked that filebeat is now deployed as daemonset. With this change now the tests pass

**Review by cachedout — APPROVED** [review]

File: .ci/Jenkinsfile [hunk 2]
```diff
@@ -47,9 +47,9 @@ pipeline {
     choice(name: 'LOG_LEVEL', choices: ['DEBUG', 'TRACE', 'INFO'], description: 'Log level to be used')
     choice(name: 'TIMEOUT_FACTOR', choices: ['5', '3', '7', '11'], description: 'Max number of minutes for timeout backoff strategies')
     string(name: 'STACK_VERSION', defaultValue: '8.0.0-SNAPSHOT', description: 'SemVer version of the stack to be used for the tests.')
-    string(name: 'HELM_CHART_VERSION', defaultValue: '7.10.0', description: 'SemVer version of Helm chart to be used.')
-    string(name: 'HELM_VERSION', defaultValue: '3.4.1', description: 'SemVer version of Helm to be used.')
-    string(name: 'HELM_KIND_VERSION', defaultValue: '0.8.1', description: 'SemVer version of Kind to be used.')
+    string(name: 'HELM_CHART_VERSION', defaultValue: '7.11.1', description: 'SemVer version of Helm chart to be used.')
```
Comment (jmlrt) [inline .ci/Jenkinsfile:50]: 7.11.2 has been released last week

**Review by jmlrt — APPROVED** [review]
LGTM⛴

File: .ci/Jenkinsfile (hunk 2 above)
Comment (mdelapenya) [reply .ci/Jenkinsfile:50]: Cool, let's update this PR then

Comment (mdelapenya) [inline .ci/Jenkinsfile:50]:
```suggestion
    string(name: 'HELM_CHART_VERSION', defaultValue: '7.11.2', description: 'SemVer version of Helm chart to be used.')
```

File: e2e/_suites/helm/README.md [hunk 3]
```diff
@@ -33,15 +33,15 @@ This is an example of the optional configuration:
 
    ```shell
    # Depending on the versions used,
-   export HELM_VERSION="3.4.1"        # Helm version: for Helm v2.x.x we have to initialise Tiller right after the k8s cluster
-   export HELM_CHART_VERSION="7.10.0"  # version of the Elastic's Observability Helm charts
+   export HELM_VERSION="3.5.2"        # Helm version: for Helm v2.x.x we have to initialise Tiller right after the k8s cluster
+   export HELM_CHART_VERSION="7.11.1"  # version of the Elastic's Observability Helm charts
```
Comment (mdelapenya) [inline e2e/_suites/helm/README.md:37]:
```suggestion
   export HELM_CHART_VERSION="7.11.2"  # version of the Elastic's Observability Helm charts
```

File: e2e/_suites/helm/helm_charts_test.go [hunk 4]
```diff
@@ -46,7 +46,7 @@ var kubectl k8s.Kubectl
 var helmVersion = "3.x"
 
 // helmChartVersion represents the default version used for the Elastic Helm charts
-var helmChartVersion = "7.10.0"
+var helmChartVersion = "7.11.1"
```
Comment (mdelapenya) [inline e2e/_suites/helm/helm_charts_test.go:49]:
```suggestion
var helmChartVersion = "7.11.2"
```

**Review by jmlrt — APPROVED** [review]
LGTM⛴

**Pull request merged**